### DISCUSSION TITLE
nativeUpdate not trigger beforeUpdate event subscripber ?

### DISCUSSION THREAD

**Discussion by superiums** [mikro-orm/mikro-orm#4567] [General]
service:
```
    update<T extends object>(mo: EntityName<T>, where:FilterQuery<T>, data?: object, options?: NativeInsertUpdateOptions<object>) {
        const repo = this.em.getRepository(mo)
         return        repo.nativeUpdate(where,data,options)
    }
```

subscriber:
```
async beforeUpdate(args: EventArgs<org>): Promise<void> {
        console.log('ori-org', args.changeSet.originalEntity)
}
```

is it a bug or i did something wrong ?

**Comment by B4nan** [comment]
Hooks are only for the operations triggered via unit of work - so when you flush, not when you use the native methods, those are basically just shortcuts to query builder. When there is no entity, there is no hook.

**Reply by superiums** [reply]
then how should i use uow to update an entity ?
as i see, there's only `upsert` in uow, no `update` avaluable.  and also i noticed there's a `upsert hook` for it.

**Reply by B4nan** [reply]
With uow you load entity, modify it and call flush. 

https://github.com/mikro-orm/guide#persist-and-flush

**Reply by superiums** [reply]
persist and flush is a create operation.
load , modify and flush , is not a common case for web, we just update the entity while receive it from client-side, not from db.
this mean we need to load it again from db ?

**Reply by B4nan** [reply]
As I said, if you want hooks, you need to do it this way, with `nativeUpdate` there is no entity instance => no hooks. Hooks are handled by the UoW, in other words, they are executed by the `flush` method.

In latest versions, you don't need to load the entity, as entity references are also tracked for changes:

```ts
const user = em.getReference(User, 123);
user.name = 'new';
await em.flush();
```

**Comment by superiums** [comment]
thanks, that's greate!
